core/src/main/java/com/linecorp/armeria/server/VirtualHost.java

```diff
@@ -195,10 +201,41 @@ private static boolean needsNormalization(String hostnamePattern) {
     }
 
     @Nullable
-    static SslContext validateSslContext(@Nullable SslContext sslContext) {
+    static SslContext validateSslContext(@Nullable SslContext sslContext) throws SSLException {
         if (sslContext != null && !sslContext.isServer()) {
             throw new IllegalArgumentException("sslContext: " + sslContext + " (expected: server context)");
         }
+
+        SSLEngine serverEngine = null;
+        SSLEngine clientEngine = null;
+
+        try {
+            serverEngine = sslContext.newEngine(ByteBufAllocator.DEFAULT);
+            serverEngine.setUseClientMode(false);
+            serverEngine.setNeedClientAuth(false);
+
+            final SslContext sslContextClient =
+                    VirtualHostBuilder.buildSslContext(SslContextBuilder::forClient, sslContextBuilder -> {});
+            clientEngine = sslContextClient.newEngine(ByteBufAllocator.DEFAULT);
+            clientEngine.setUseClientMode(true);
+
+            clientEngine.beginHandshake();
```

Comment:
Could you add a comment that why we call this method?

Reply:
Added comment on the method. PTAL :bowing_man:

Reply:
I mean why we need to call `beginHandshake();`.
And I found out from the Javadoc of `beginHandshake()`:
```java
* This method is not needed for the initial handshake, as the
* {@code wrap()} and {@code unwrap()} methods will
* implicitly call this method if handshaking has not already begun.
```
So we might not need to call this? Not sure this is right. 😄

Reply:
AFAIU, we can remove `beginHandshake()`

I searched netty's `SSLEngine` implemantation of `RefernceCountedOpenSslEngine`
and I found following,

Conclusion:
* We don't need to call `beginHandshake()` when it is first handshake procedure.

Reference source code:
* from : io.netty.handler.ssl.ReferenceCountedOpenSslEngine.java

Checks:
* Initial value of `SSLEngine.handshakeState` is `NOT_STARTED`
* `SSLEngine.beginHandshake()` set `SSLEngine.handshakeState` to `STARTED_EXPLICITLY`
``` java
case NOT_STARTED:
  handshakeState = HandshakeState.STARTED_EXPLICITLY;
  if (handshake() == NEED_TASK) {
      // Set needTask to true so getHandshakeStatus() will return the correct value.
     needTask = true;
  }
  calculateMaxWrapOverhead();
  break;
```
* If `SSLEngine` try to first handshake without calling `beginHandshake()`,
`wrap()` or `unwrap()` method set `SSLEngine.handshakeState` to `STARTED_EXPLICITLY`
``` java
// wrap()
// Flush any data that may be implicitly generated by OpenSSL (handshake, close, etc..).
  SSLEngineResult.HandshakeStatus status = NOT_HANDSHAKING;
  // Prepare OpenSSL to work in server mode and receive handshake
  if (handshakeState != HandshakeState.FINISHED) {
      if (handshakeState != HandshakeState.STARTED_EXPLICITLY) {
          // Update accepted so we know we triggered the handshake via wrap
          handshakeState = HandshakeState.STARTED_IMPLICITLY;
      }
```
``` java
// unwrap()
SSLEngineResult.HandshakeStatus status = NOT_HANDSHAKING;
  // Prepare OpenSSL to work in server mode and receive handshake
  if (handshakeState != HandshakeState.FINISHED) {
      if (handshakeState != HandshakeState.STARTED_EXPLICITLY) {
          // Update accepted so we know we triggered the handshake via wrap
          handshakeState = HandshakeState.STARTED_IMPLICITLY;
      }
```

Reply:
I also run `validateSslContext()` with/without `beginHandshake()`.
And results are same.

So, I think we can remove `beginHandshake()` in this case.
Let me submit fixed commit.